core/block.go

```diff
@@ -39,6 +39,16 @@ type Header struct {
 	Signatures [][]*felt.Felt
 	// Amount of STRK charged per Gas spent
 	GasPriceSTRK *felt.Felt
+	// The mode of the L1 data availability
+	L1DAMode string
+	// The gas price for L1 data availability
+	L1DataGasPrice GasPrice
+	// The gas price for L1
+	L1GasPrice GasPrice
```

Comment:
these won't be available for old block, we can make them pointers

```suggestion
	L1DataGasPrice *GasPrice
	// The gas price for L1
	L1GasPrice *GasPrice
```